node-notes-cli/solution.js

```diff
@@ -0,0 +1,39 @@
+const $json = require('./data.json');
+const fs = require('fs');
+
+if (process.argv[2] === 'what') {
+  $json.nextId++;
+}
+
+if (process.argv[2] === 'read') {
+  let i = 0;
+  const keys = Object.keys($json.notes);
+  while (i < keys.length) {
+    for (const key in $json.notes) {
+      console.log(keys[i] + ': ' + JSON.stringify($json.notes[key], null, 2));
+      i++;
+    }
+  }
```

Comment:
You can simplify this a bit by removing the keys array and leaving out the while loop

```js
for (const key in $json.notes) {
  console.log(key + ':', $json.notes[key]);
}
```